.github/workflows/ci-spread-times.yaml

```
@@ -66,3 +67,137 @@ jobs:
+
+  avg-task-times:
+    runs-on: ["self-hosted", "spread-enabled"]
+    steps:
+      - name: Cleanup job workspace
+        id: cleanup-job-workspace
+        run: |
+          rm -rf "${{ github.workspace }}"
+          mkdir "${{ github.workspace }}"
+
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          # spread uses tags as delta reference
+          fetch-depth: 0
+
+      - name: create new empty test
+        run: |
+          mkdir tests/smoke/empty
+          echo -e "summary: test\nexecute: true\n" > tests/smoke/empty/task.yaml
+
+      - name: run repeated empty test with spread
+        env:
+          SPREAD_USE_SNAPD_SNAP_URL: https://storage.googleapis.com/snapd-spread-tests/snapd-tests/snaps/snapd_master_amd64.snap
+          SPREAD_USE_PREBUILT_SNAPD_SNAP: true
+        run: |
+          # Run spread several times to get the avg time for the prepare/restore task
+          # Test task is executed just in the some systems
+          spread -no-debug-output -repeat 10 -json report.json openstack-arm-ps7:ubuntu-24.04-arm-64:tests/smoke/empty openstack-ps7:ubuntu-24.04-64:tests/smoke/empty openstack-ps7:ubuntu-core-24-64:tests/smoke/empty openstack-ps7:arch-linux-64:tests/smoke/empty openstack-ps7:amazon-linux-2023-64:tests/smoke/empty
+
+      - name: Write the metrics to prometheus
+        if: always()
+        env:
+          METRICS_HOST: ${{ secrets.METRICS_HOST }}
+          METRICS_PORT: ${{ secrets.METRICS_PORT }}
+        run: |
+          json_file="report.json"
+
+          if [ ! -f "$json_file" ]; then
+              echo "The report file doesn't exist"
+              exit 1
+          fi
+
+          # Step 1: Add duration_ms to each item
+          # To calculate the duration we use 
+          #   sub: command to remove the fractional seconds
+          #   strptime: command to parse the string into a structured date object
+          #   mktime: converts that structured date into seconds
+          #   capture: is used to retrieve the milliseconds part of the date string
+          #   * 1000: converts seconds to milliseconds
+          jq -r '
+            .items[]
+            | select(
+                .success == true
+                and (.verb == "preparing" or .verb == "restoring")
+                and .level == "task"
+              )
+            | {
+                verb,
+                backend,
+                system,
+                level,
+                name,
+                duration: (
+                  (
+                    ((.end | sub("\\.[0-9]+";"") | strptime("%Y-%m-%dT%H:%M:%S") | mktime) * 1000)
+                  + ((.end | capture("\\.(?<ms>[0-9]+)") | .ms | tonumber))
+                  -
+                    ((.start | sub("\\.[0-9]+";"") | strptime("%Y-%m-%dT%H:%M:%S") | mktime) * 1000)
+                  - ((.start | capture("\\.(?<ms>[0-9]+)") | .ms | tonumber))
```

Shouldn't this be `+`? Aren't you summing together the milliseconds part with the date, hour, minute, and seconds?
```suggestion
                  + ((.start | capture("\\.(?<ms>[0-9]+)") | .ms | tonumber))
```

---

I think this is ok, see thes examples: 

end: a.555 start a.444
we have a + 555 - a - 444 = 555 - 444 = 111

end: a.555 start (a-2000).444
we have a + 555 - ( a - 2000 ) - 444 = 555 + 2000 - 444 = 2111

---

Ah sorry, yes you're right. I misread the parentheses